khiops/sklearn/estimators.py

```
@@ -384,13 +383,13 @@ def fit(self, X, y=None, **kwargs):
         # If on "fitted" state then:
         # - self.model_ must be a DictionaryDomain
         # - self.model_report_ must be a KhiopsJSONObject
-        if hasattr(self, "is_fitted_") and self.is_fitted_:
-            assert hasattr(self, "model_") and isinstance(
-                self.model_, kh.DictionaryDomain
-            )
-            assert hasattr(self, "model_report_") and isinstance(
-                self.model_report_, kh.KhiopsJSONObject
-            )
+        try:
+            check_is_fitted(self)
+            assert isinstance(self.model_, kh.DictionaryDomain)
+            assert isinstance(self.model_report_, kh.KhiopsJSONObject)
+            assert isinstance(self.model_, kh.DictionaryDomain)
+        except NotFittedError:
+            pass
```

I would explain in a comment here why we ignore the exception.

---

Ok & Done.